### PR TITLE
[NOTIFICATIONS #6] refactor(orders): bind expired order notification to events emitter

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Following the standard pattern I've bound order expiration notification to event emitter.
Please, check other PRs in the waterfall, because I've significantly refactored this.

image

  # To Test

See https://github.com/cowprotocol/cowswap/pull/3934

### CODE REVIEW

File: apps/cowswap-frontend/src/legacy/state/orders/middleware/batchExpireOrdersPopup.ts [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
-export function batchExpireOrdersPopup(
-  store: MiddlewareAPI<Dispatch, AppState>,
-  payload: ExpireOrdersBatchParams,
-  orders: OrdersStateNetwork
-) {
-  payload.ids.forEach((id) => {
-    const orderObject = getOrderByIdFromState(orders, id)
-
-    // Do not trigger expired pop up if order is hidden
-    if (orderObject && !orderObject.order.isHidden) {
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/state/orders/middleware/batchExpireOrdersPopup.ts:25]: Is the hidden logic captured anywhere?
I don't see it handled anywhere else in this PR

File: apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  if (!order) return
+
+  const tx = {
+    hash: orderUid,
+    hashType: HashType.ETHEREUM_TX,
+  }
+
+  // TODO: do we need this?
+  // orderAnalytics('Expired', getUiOrderType(orderObject.order))
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx:44]: Same comment as the other PR, only if we care about analytics events in general

File: apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    onToastMessage,
+  } = props
+
+  const ref = useCallback(
+    (node: HTMLDivElement) => {
+      if (node) onToastMessage(node.innerText)
+    },
+    [onToastMessage]
+  )
```
Comment (anxolin) [inline apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx:32]: this pattern seems a bit obscure to me. If we are using it, at least we could make a reusable hook

**Review by anxolin — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx (hunk 3 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx:32]: Agree with you, refactored in https://github.com/cowprotocol/cowswap/pull/3934

File: apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx (hunk 2 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx:44]: I will rise a discussion about it, thank you!

**Pull request merged**